exercises/pangram/canonical-data.json

```diff
@@ -12,57 +12,81 @@
       ],
       "cases": [
         {
-          "description": "sentence empty",
+          "description": "can handle an empty sentence",
           "property": "isPangram",
           "input": "",
           "expected": false
         },
         {
-          "description": "pangram with only lower case",
+          "description": "handles an undefined message as empty message",
           "property": "isPangram",
-          "input": "the quick brown fox jumps over the lazy dog",
+          "input": "null",
+          "expected": false
+        },
+        {
+          "description": "recognizes a lower case pangram",
+          "property": "isPangram",
+          "input": "thequickbrownfoxjumpsoverthelazydog",
           "expected": true
         },
         {
-          "description": "missing character 'x'",
+          "description": "recognizes a missing character 'x'",
+          "property": "isPangram",
+          "input": "aquickmovementoftheenemywilljeopardizefivegunboats",
+          "expected": false
+        },
+        {
+          "description": "recognizes a missing character 'h'",
           "property": "isPangram",
-          "input": "a quick movement of the enemy will jeopardize five gunboats",
+          "input": "fiveboxingwizardsjumpquicklyatit",
           "expected": false
         },
         {
-          "description": "another missing character 'x'",
+          "description": "recognizes an upper case pangram",
           "property": "isPangram",
-          "input": "the quick brown fish jumps over the lazy dog",
+          "input": "THEQUICKBROWNFOXJUMPSOVERTHELAZYDOG",
+          "expected": true
+        },
+        {
+          "description": "recognizes a missing character in upper case",
+          "property": "isPangram",
+          "input": "THEQUICKBROWNFOXJUMPSOVERTHELAZYDOLL",
           "expected": false
         },
         {
-          "description": "pangram with underscores",
+          "description": "recognizes a mixed case pangram",
           "property": "isPangram",
-          "input": "the_quick_brown_fox_jumps_over_the_lazy_dog",
+          "input": "ThEFiVebOxInGWiZaRdSJuMpqUiCkLy",
           "expected": true
         },
         {
-          "description": "pangram with numbers",
+          "description": "recognizes a missing character in mixed case",
+          "property": "isPangram",
+          "input": "ThEFiVebOxInGWiZaRdSJuMpqUiCkL",
+          "expected": false
+        },
+        {
+          "description": "ignores other characters in correct pangram",
```

Comment:
"correct" is not the right word here - it's a pangram or a non-pangram.

Reply:
I agree, but because of the context I wrote it extra clear to make it distinct to the complementary non-pangram input. Sometimes I feel it might be better to be redundantly clear instead.